lib/charms/layer/apache_bigtop_base.py

```diff
@@ -68,6 +71,9 @@ def apply_patches(self):
             utils.run_as('root', 'patch', '-p1', '-s', '-i', client_patch)
             # and finally, patch site.pp to skip jdk install
             utils.run_as('root', 'patch', '-p1', '-s', '-i', java_patch)
+            # apply configurable spark logs patch
+            utils.run_as('root', 'patch', '-p0', '-s', '-i', spark_logs_patch)
```

Comment:
I was cheating here :) the patch did not have the root directory. But using the -p1 flag is better, so I updated the patch and the source.